src/Pagination.js

```diff
@@ -1,51 +1,44 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import cx from 'classnames';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
 import Icon from './Icon';
 import PaginationButton from './PaginationButton';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
 
-class Pagination extends Component {
-  constructor(props) {
-    super(props);
-    const { activePage, items } = props;
-
-    this.state = {
-      activePage: activePage > 0 && activePage <= items ? activePage : 1
-    };
-
-    this.renderButtons = this.renderButtons.bind(this);
-    this._onClick = this._onClick.bind(this);
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.activePage !== this.props.activePage) {
-      this.setState({ activePage: nextProps.activePage });
+const Pagination = props => {
+  const {
+    items,
+    className,
+    children,
+    maxButtons = items,
+    leftBtn,
+    rightBtn,
+    onSelect
+  } = props;
+
+  const [activePage, setActivePage] = useState(
+    props.activePage > 0 && props.activePage <= items ? props.activePage : 1
+  );
+
+  useEffect(() => {
+    if (activePage !== props.activePage) {
+      setActivePage(props.activePage);
     }
-  }
-
-  _onClick(i) {
-    const { items, onSelect } = this.props;
+  }, [activePage, props.activePage]);
 
-    return () => {
+  const onClick = useCallback(
+    i => () => {
```

Comment:
can we convert this to a single fn? `const onClick = useCallback(i => {...` and consume it on renders like so `onSelect={() => onClick(i)}`